torch/_dynamo/config.py

```diff
@@ -220,12 +220,19 @@
 # false_fn produces code with identical guards.
 enforce_cond_guards_match = True
 
-# Automatically split model graph into pieces to match DDP bucket sizes
-# to allow DDP comm/compute overlap.  Disable to allow DDP models to
-# run without graph-breaks, but also without comm/compute overlap.
-# set TORCH_LOGS env to include any of 'dynamo', 'distributed', or
+# The flag to specify how to optimize a DDP module when it is compiled.
+# If the flag is set to True, Dynamo will automatically split model graph into
+# pieces to match DDP bucket sizes to allow DDP comm/compute overlap. Setting
+# the flag to False will allow DDP models to run without graph-breaks, but
+# also without comm/compute overlap.
+# Set TORCH_LOGS env to include any of 'dynamo', 'distributed', or
 # 'dist_ddp' for more info about optimize_ddp behavior.
-optimize_ddp = True
+#
+# Another optimization mode requires the usage of compiled_autograd. When the
+# flag is 'python_reducer', DDP will disable C++ reducer and use Python reducer
+# to allow compiled_autograd to trace the communication and allow comm/compute
+# overlap without graph-breaks.
+optimize_ddp: Union[bool, str] = True
```

Comment:
sigh.. 
config used to be an object.  so you could have setters/getters and some control over the format.  In my head I was envisioning that when the user sets the value (True/False) we'd parse that, convert it to the right enum value, and then getter would always produce enum.  Then in the code that checks the config we just rely on enum for less confusion.

since this is not the case anymore, i think the next best thing is to define a getter that we use from the code.  

config.get_optimize_ddp_mode() -> enum (or str if you prefer) which will do the type checking and normalize the true/false into one of the supported modes so the downstream code can always use the normalized way

Reply:
I prefer to convert the result to an enum but torch will complain having an enum in a config file. So I'll convert the boolean to str.

Reply:
latest version lgtm.  Except, i was not sure why you need `m = sys.modules[__name__]`, instead of just accessing `optimize_ddp` ?